### PR TITLE
[Refactor] Centralize logic of global electrode id 

### PR DESCRIPTION
**Pull request by h-mayorquin**
Refactor of `add_electrodes_table`. I will push fixes for #958 and #982 with this.

Basically, I am defining the mapping betweeen the recording and the electrodes table in a centralized place instead of having it decoupled in two different places as we have in main.

There is nothing new added. I moved one test a bit and that shows in the diff but nothing is new.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Uh, some of the failing tests here are passing locally for me. This is strange.

**Comment by h-mayorquin** [comment]
OK, I discovered what I think is failing here:

https://github.com/catalystneuro/neuroconv/blob/31fca672d8836b25634e9cdb753828df0eec754f/src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglx_utils.py#L18-L23

Sometimes the property "shank electrode number"  is int and sometimes the property is str. That does no texplain the odd pattern of failing with python version but should be fix.

**Comment by h-mayorquin** [comment]
OK, this is ready, it was an condition inside and if loop that I did not handle well.

File: pyproject.toml [hunk 1]
```diff
@@ -30,8 +30,8 @@ classifiers = [
 ]
 requires-python = ">=3.9"
 dependencies = [
-    "numpy>=1.22.0; python_version <= '3.11'",
-    "numpy>=1.26.0; python_version >= '3.12'",
+    "numpy>=1.22.0, <2.0.0; python_version <= '3.11'",
```
Comment (CodyCBakerPhD) [inline pyproject.toml:33]: What is status on 2.0 support?

**Review by CodyCBakerPhD — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (h-mayorquin) [reply pyproject.toml:33]: I think pynwb recently added it, right? I added it to most of spikeinterface but just being cautios here.

Comment (CodyCBakerPhD) [reply pyproject.toml:33]: I thought I heard they fixed it

**Pull request merged**